core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop = NULL;
+	int len_prop = 0;
+	int it_num = -1;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if (!int_prop || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
+						const char *compatible)
+{
+	paddr_t offset = 0;
+	int fnode = 0;
+	int node_status = 0;
+
+	fnode = fdt_node_offset_by_compatible(fdt, *node, compatible);
+
+	while (fnode != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, fnode);
+		if ((node_status & status) == status) {
+			offset = _fdt_reg_base_address(fdt, fnode);
+			*node  = fnode;
+			break;
+		}
+
+		fnode = fdt_node_offset_by_compatible(fdt, fnode, compatible);
+	}
+
+	return offset;
+}
+
+int dt_disable_status(void *fdt, int node)
+{
+	const char *prop = NULL;
+	int len = 0;
+
+	prop = fdt_getprop(fdt, node, "status", &len);
+	if (!prop) {
+		/* Status is not available, just add it */
+		if (fdt_setprop_string(fdt, node, "status", "disabled"))
+			return -1;
+	} else {
+		/*
+		 * Status is there, modify it.
+		 * Ask to set "disabled" value to the property. The value
+		 * will be automatically truncated with "len" size by the
+		 * fdt_setprop_inplace function.
+		 * Setting a value different from "ok" or "okay" will disable
+		 * the property.
+		 * Setting a truncated value of "disabled" with the original
+		 * property "len" is preferred to not increase the DT size and
+		 * losing time in recalculating the overall DT offsets.
+		 * If original length of the status property is larger than
+		 * "disabled", the property will start with "disabled" and be
+		 * completed with the rest of the original property.
+		 */
+		if (fdt_setprop_inplace(fdt, node, "status", "disabled", len))
+			return -1;
+	}
+
+	return 0;
+}
+
+int dt_set_secure_status(void *fdt, int node)
+{
+	if (dt_disable_status(fdt, node)) {
+		EMSG("Unable to disable Normal Status");
+		return (-1);
+	}
+
+	/*
+	 * Append the "secure-status" property in the DT
+	 * If the property is not present, it will be added
+	 * otherwise it will be modified to be "okay"
+	 */
+	if (fdt_appendprop_string(fdt, node, "secure-status", "okay"))
```

Comment:
Looks like this sequence could be used in `dt_disable_status()`, no?

Reply:
no, the dt_disable_status sets the property `status` and the dt_set_secure_status sets the property `secure-status` used in the function _fdt_get_status.

Reply:
I agree. My point is, why not using the following implementation?
```C
int dt_disable_status(void *fdt, int node) 
{
	if (fdt_appendprop_string(fdt, node, "status", "disabled"))
		return -1;

	return 0;
}

int dt_set_secure_status(void *fdt, int node)
{
	if (dt_disable_status(fdt, node))
		return -1;

	if (fdt_appendprop_string(fdt, node, "secure-status", "okay"))
		return -1;

	return 0;
}

Reply:
After test it seems that the `appendprop` is not the correct function to use in both cases.
The `appendprop` appends the new status value to the current existing property and doesn't modify it.
Hence I'm keeping the current dt_disable_status function that seems to be better in the sense  that the property `status` is added if not existing otherwise it modify it to be something else than `okay` or `ok`. When the property size is not changed, we are not spending time in recalculating the overall DT offsets.
I'll change the `dt_set_secure_status` to `dt_enable_secure_status` and use the function `fdt_setprop_string` in place of the `fdt_appendprop_string`. In this case we want to have the `secure-status` set to `okay` to enable it.